contracts/ColonyStorage.sol

```diff
@@ -61,13 +62,13 @@ contract ColonyStorage is DSAuth {
   mapping (uint256 => RewardPayoutCycle) rewardPayoutCycles;
   // Active payouts for particular token address. Assures that one token is used for only one active payout
   mapping (address => bool) activeRewardPayouts;
-  // Incremented every time a reward payout is created. Used for comparing with `userRewardPayoutCount`
-  // to ensure that rewards are claimed in the right order.
+  // Incremented every time a reward payout is created. Used for indexing the reward payout
   // Can be used for iterating over all reward payouts (excluding index 0)
   uint256 globalRewardPayoutCount;
-  // Keeps track of how many payouts are claimed by the particular user.
-  // Can be incremented either by waiving or claiming the reward.
-  mapping (address => uint256) userRewardPayoutCount;
+
+  // Keeps track if user claimed or waived specific reward payout.
+  // Used for ensuring that user cannot claim or waive the reward twice
+  mapping (address => mapping(uint256 => bool)) userRewardPayouts;
```

Comment:
I don't yet understand this change but it maybe related to the double spending you were talking about in justifying `lastDepositTimestamp`

Reply:
This will be simply achieved with setting the user lock count to the global one `userLockCount = totalLockCount` won't it?

Reply:
No, the payout counters are different thing and needs to be handled separately.

Reply:
ok yes you're right